docs/standard/exceptions/best-practices-for-exceptions.md

```diff
@@ -1,129 +1,74 @@
+## Handling exceptions
+
+The following best practices concern how you handle exceptions:
+
+- [Use try/catch/finally blocks to recover from errors or release resources](#use-trycatchfinally-blocks-to-recover-from-errors-or-release-resources)
+- [Handle common conditions to avoid exceptions](#handle-common-conditions-to-avoid-exceptions)
+- [Catch cancellation and asynchronous exceptions](#catch-cancellation-and-asynchronous-exceptions)
+- [Design classes so that exceptions can be avoided](#design-classes-so-that-exceptions-can-be-avoided)
+- [Restore state when methods don't complete due to exceptions](#restore-state-when-methods-dont-complete-due-to-exceptions)
+- [Capture exceptions to rethrow later](#capture-exceptions-to-rethrow-later)
+
+### Use try/catch/finally blocks to recover from errors or release resources
+
+For code that can potentially generate an exception, and when your app can recover from that exception, use `try`/`catch` blocks around the code. In `catch` blocks, always order exceptions from the most derived to the least derived. (All exceptions derive from the <xref:System.Exception> class. More derived exceptions aren't handled by a `catch` clause that's preceded by a `catch` clause for a base exception class.) When your code can't recover from an exception, don't catch that exception. Enable methods further up the call stack to recover if possible.
 
 Clean up resources that are allocated with either `using` statements or `finally` blocks. Prefer `using` statements to automatically clean up resources when exceptions are thrown. Use `finally` blocks to clean up resources that don't implement <xref:System.IDisposable>. Code in a `finally` clause is almost always executed even when exceptions are thrown.
 
-## Handle common conditions without throwing exceptions
+### Handle common conditions to avoid exceptions
 
-For conditions that are likely to occur but might trigger an exception, consider handling them in a way that will avoid the exception. For example, if you try to close a connection that's already closed, you'll get an `InvalidOperationException`. You can avoid that by using an `if` statement to check the connection state before trying to close it.
+For conditions that are likely to occur but might trigger an exception, consider handling them in a way that avoids the exception. For example, if you try to close a connection that's already closed, you'll get an `InvalidOperationException`. You can avoid that by using an `if` statement to check the connection state before trying to close it.
```

Comment:
> You can avoid that

Most of the time. In many of these situations there can be race conditions where such an up-front check will eliminate the exception 99% of the time, but if between the check and the operation the guarded condition changed, you could still incur an exception.